### PR TITLE
Update ponylang library settings

### PR DESCRIPTION
**Pull request by Theodus**
Update markdown files, CI, and Makefile to be consistent with other ponylang projects

### CODE REVIEW

File: .ci-scripts/x86-64-unknown-linux-nightly.bash [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+PACKAGE_DIR=$(mktemp -d)
+PACKAGE=ponyup-${TRIPLE}
+
+# Cloudsmith configuration
+CLOUDSMITH_VERSION=${TODAY}
+ASSET_OWNER=ponylang
+ASSET_REPO=nightlies
+ASSET_PATH=${ASSET_OWNER}/${ASSET_REPO}
+ASSET_FILE=${PACKAGE_DIR}/${PACKAGE}.tar.gz
+ASSET_SUMMARY="Pony dependency manager"
```
Comment (SeanTAllen) [inline .ci-scripts/x86-64-unknown-linux-nightly.bash]: This needs to be changed.

**Comment by SeanTAllen** [comment]
@Theodus if you check the ponyc circleci file you'll see we already have a ci image for shellcheck so you can switch to using that (and we have a standard naming scheme for those as well).

I think that can replace the shellcheck-init-script that you have.

**Comment by SeanTAllen** [comment]
If the .sh files are bash specific, I'd prefer to name them .bash

**Comment by Theodus** [comment]
> If the .sh files are bash specific, I'd prefer to name them .bash

They are not bash specific.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,35 +1,85 @@
...
+      - run: bash .ci-scripts/x86-64-unknown-linux-nightly.bash ${CLOUDSMITH_API_KEY}
+      - zulip/status
 
+  shellcheck-init-script:
+    docker:
+      - image: ponylang/ponyc:latest
+    steps:
+      - checkout
       - run: apt-get update -y && apt-get install -y shellcheck
       - run: shellcheck ponyup-init.sh
```
Comment (SeanTAllen) [inline .circleci/config.yml]: do you want this to alert on failure?

i'd say yes in which case see the zulip/status below that alerts on failure only.

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,35 +1,85 @@
...
+      - run: .ci-scripts/test-ponyup-ponyc.sh
+      - zulip/status:
+          fail_only: true
+
+  vs-ponyc-master:
+    docker:
+      - image: ponylang/ponyc:latest
+    steps:
+      - checkout
+      - run: make test config=debug
```
Comment (SeanTAllen) [inline .circleci/config.yml]: if there arent "config=debug" and "config=release" jobs for each, i'd lean towards not doing config=debug here and on the other job as well.

File: Makefile [hunk 4]
```diff
@@ -1,19 +1,102 @@
...
-	stable env $(PONYC) -D$(SSL_VERSION) -d cmd -o build -b ponyup
+ifneq ($(arch),)
+  PONYC_FLAGS += --cpu $(arch)
+endif
 
-clean:
-	rm -rf build/*
+ifdef static
+  ifeq (,$(filter $(static),true false))
+		$(error "static must be true or false)
```
Comment (SeanTAllen) [inline Makefile]: formatting looks off here.

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,35 +1,85 @@
...
       - run: apt-get update -y && apt-get install -y shellcheck
       - run: shellcheck ponyup-init.sh
 
+  vs-ponyc-release:
+    docker:
+      - image: ponylang/ponyc:release
+    steps:
+      - checkout
       - run: .ci-scripts/install-openssl.sh
+      - run: make test config=debug
```
Comment (SeanTAllen) [inline .circleci/config.yml]: you makefile requires `ssl` to be set but this isn't using that.

File: .circleci/config.yml (hunk 3 above)
Comment (SeanTAllen) [reply .circleci/config.yml]: you makefile requires `ssl` to be set but this isn't using that.

**Comment by SeanTAllen** [comment]
@Theodus they will work with any "posix shell"? dash for example?

**Comment by Theodus** [comment]
> @Theodus they will work with any "posix shell"? dash for example?

CI has been testing on Ubuntu, which runs these with `/bin/sh` which is dash by default. Shellcheck also agrees that these are suitable for posix shells.

**Comment by SeanTAllen** [comment]
Awesome

File: .ci-scripts/x86-64-unknown-linux-nightly.bash [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+PACKAGE_DIR=$(mktemp -d)
+PACKAGE=ponyup-${TRIPLE}
+
+# Cloudsmith configuration
+CLOUDSMITH_VERSION=${TODAY}
+ASSET_OWNER=ponylang
+ASSET_REPO=nightlies
+ASSET_PATH=${ASSET_OWNER}/${ASSET_REPO}
+ASSET_FILE=${PACKAGE_DIR}/${PACKAGE}.tar.gz
+ASSET_SUMMARY="The Pony toolchain multiplexer"
```
Comment (SeanTAllen) [inline .ci-scripts/x86-64-unknown-linux-nightly.bash]: Love the summary!

**Pull request merged**